modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/reader/BatchEngineImportTaskItemReaderUtil.java

```diff
@@ -60,13 +62,29 @@ public static <T> T convertValue(
 				}
 			}
 
+			JsonDeserialize[] jsonDeserialize = field.getAnnotationsByType(
+				JsonDeserialize.class);
+
+			Object value = entry.getValue();
+
+			if ((jsonDeserialize.length > 0) &&
+				!Objects.equals(value.getClass(), String.class)) {
+
+				try {
+					value = _objectMapper.writeValueAsString(value);
+				}
+				catch (JsonProcessingException jsonProcessingException) {
+					if (_log.isDebugEnabled()) {
```

Comment:
What if there is an error? Should it fail? or which is the expected behaviour in this case?

Reply:
no longer applies.